haskell-language-server.cabal

```
@@ -340,6 +340,8 @@ test-suite hls-class-plugin-tests
   import:           defaults, pedantic, test-defaults, warnings
   if !flag(class)
     buildable: False
+  if impl(ghc >= 9.8)
+    ghc-options: -Wwarn=x-partial
```

Given that @fendor is for using partial functions in tests, I'm even considering switching to disabling these warnings for tests as there's no point in making the build output more noisy for non-actionable stuff (My favorite [broken window theory](https://en.wikipedia.org/wiki/Broken_windows_theory) - the more warnings / flaky tests there are, the more likely people are to ignore them).
```suggestion
    ghc-options: -Wno-x-partial
```

---

Maybe I can even put it into `common test-defaults` as there's no reason to have additional mechanism to prevent new occurrences (we already have hlint in ci)

---

Sounds like a great idea!